hca/parser.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import argparse
+import pprint
+
+
+def make_name(http_method, path_split):
+    """Name an endpoint."""
+    # If the api needs file/write/{sdf} functionality, will become put-file-write
+    name = [http_method]
+    path_non_args = list(filter(lambda x: len(x) > 0 and x[0] != "{", path_split))
+    for non_arg in path_non_args:
+        name.append(non_arg)
+    name = "-".join(name)
+    return name
+
+
+def index_parameters(endpoint_info):
+    """Map all parameters from parameter name -> parameter info."""
+    indexed_parameters = {}
+
+    if "parameters" not in endpoint_info:
+        return indexed_parameters
+
+    for param in endpoint_info['parameters']:
+        if "schema" in param:
+            # upper_required = param['required']
+            schema_params = param['schema']['properties']
+            for schema_param_name in schema_params:
+                param_name = param['name'] + "-" + schema_param_name
+                payload = schema_params[schema_param_name]
+                payload["name"] = param_name
+                payload['required'] = False  # Placeholder before checking required array
+                payload['in'] = param["in"]
+                indexed_parameters[param_name] = payload
+            for required_name in param['schema']['required']:
+                param_name = param['name'] + "-" + required_name
+                indexed_parameters[param_name]['required'] = True
+        else:
+            param_name = param['name']
+            indexed_parameters[param_name] = param
+    # pprint.pprint(indexed_parameters)
+    return indexed_parameters
+
+
+def get_or_else(d, key, els):
+    """Get key from d. If key not in d, get els."""
+    return d[key] if key in d else els
```

Comment:
I suggest dropping this function. It's a one-liner that is better written as `d.get(key, els)`, and it has a docstring that adds no information.